### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by amar-hunar**
This PR fixes bugs and adds props in some components

## Features:

The following bugfixes are done:

- selected options bug in `CustomSelect`
- update `onBlur` behavior in `EditableTextField`, `EditableTextArea` (when the field has error and onBlur occurs, reset the field to last saved value)

The following updates are done:

- add `menuSx` and `menuMarginThreshold` props in `CustomSelect` and `MobileDatePicker`
- update `CustomSelect`, `MobileDatePicker` stories

## Links:
- [Storybook Link](https://bugfixes--666ac08efe69dd9c59a1e4c6.chromatic.com/?path=/story/components-customselect--single-select)

### CODE REVIEW

File: src/components/CustomSelect/CustomSelect.stories.tsx [hunk 1]
```diff
@@ -139,6 +141,15 @@ const CustomSelectStates = (props: CustomSelectProps) => {
                 error
                 helperText={<HelperText hasError errorMsg="I have error" />}
             />
+            <CustomSelectSection
+                sectionTitle="Custom Menu Placement (Desktop Only)"
+                // eslint-disable-next-line max-len
+                sectionDescription="Use `anchorOrigin`, `transformOrigin` and `menuMarginThreshold` to customize how the menu is placed"
+                {...props}
```
Comment (misss-popcorn) [inline src/components/CustomSelect/CustomSelect.stories.tsx:148]: can we add props in the end?

File: src/components/MobileDatePicker/MobileDatePicker.stories.tsx [hunk 2]
```diff
@@ -150,6 +152,15 @@ const MobileDatePickerStates = (props: MobileDatePickerProps) => {
                 error
                 helperText={<HelperText hasError errorMsg="I have error" />}
             />
+            <MobileDatePickerSection
+                sectionTitle="Custom Menu Placement (Desktop Only)"
+                // eslint-disable-next-line max-len
+                sectionDescription="Use `anchorOrigin`, `transformOrigin` and `menuMarginThreshold` to customize how the menu is placed"
+                {...props}
```
Comment (misss-popcorn) [inline src/components/MobileDatePicker/MobileDatePicker.stories.tsx:159]: can we move {...props} in the end?

File: src/components/CustomSelect/CustomSelect.stories.tsx (hunk 1 above)
Comment (amar-hunar) [reply src/components/CustomSelect/CustomSelect.stories.tsx:148]: This is intentional. The prop passed after `{...props}` will override the values coming by default in the story.

File: src/components/MobileDatePicker/MobileDatePicker.stories.tsx (hunk 2 above)
Comment (amar-hunar) [reply src/components/MobileDatePicker/MobileDatePicker.stories.tsx:159]: This is intentional. The prop passed after `{...props}` will override the values coming by default in the story. We cannot move `{...props}` to the end

**Pull request merged**